Trifolia.Web/Scripts/TemplateEdit/TemplateEditViewModel.js

```diff
@@ -24,6 +24,113 @@ var templateEditViewModel = function (templateId, defaults) {
     self.AvailableExtensions = ko.observableArray([]);
     self.SelectedAvailableExtensionId = ko.observable();
 
+    self.duplicateNode = function () {
+        if (!self.CurrentNode() || !self.CurrentNode().Constraint()) {
+            return false;
+        }
+
+        var constraint = self.CurrentNode().Constraint();
+
+        if (!constraint.IsPrimitive()) {
+            for (var x = 0; x < self.Constraints().length; x++) {
+                if (constraint.Context() === self.Constraints()[x].Context() && constraint.Id() !== self.Constraints()[x].Id()) {
+                    return true;
+                }
+            }
+        } else {
+            var siblings = constraint.Parent().Children();
+            for (var x = 0; x < siblings.length; x++) {
+                if (constraint.Context() === siblings[x].Context() && constraint.Id() !== siblings[x].Id()) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    self.showMoveUp = function () {
```

Comment:
As good practice, we should have JSDoc comments on each of these public methods.
I have not been very good at it, either. In these cases, they will be simple... Just a description, no parameters. See http://usejsdoc.org/ for syntax on JSDoc if needed.